core/embed/trezorhal/stm32f4/syscall_verifiers.c

```diff
@@ -0,0 +1,667 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "common.h"
+#include "syscall_probe.h"
+#include "syscall_verifiers.h"
+
+#ifdef SYSCALL_DISPATCH
+
+// ---------------------------------------------------------------------
+
+void systask_exit_error__verified(systask_t *task, const char *title,
+                                  size_t title_len, const char *message,
+                                  size_t message_len, const char *footer,
+                                  size_t footer_len) {
+  char title_copy[64] = {0};
+  char message_copy[64] = {0};
+  char footer_copy[64] = {0};
+
+  title_len = title ? MIN(title_len, sizeof(title_copy) - 1) : 0;
```

Comment:
I considered this as well and ultimately decided on this solution. In general, we shouldn’t clip the string, but I think, in this case, it’s better to display a clipped (but still relevant) message rather than a generic one. Additionally, this behavior is consistent with calling these functions directly (the string gets clipped in that case too). Maybe we could just update the comment in the header for the `system_exit_xxx` functions. How about this?

Reply:
I still think that if an applet attempts access violation, we should show that and properly respond to that by killing the applet, potentially blacklisting it or something, rather than forcing the syscall into boundaries. In other words, access violation has higher priority than whatever message the applet wants to show.

You can't get access violation when calling these functions directly, so this comparison seems a bit irrelevant. 

You would still clip the string if the applet calls this with larger than 64, but fully accessible, string.

Reply:
I finally got you point, see this fix c338491841d2643167812c79a1aa0977bd8ad02a